charts/ocis/templates/_common/_tplvalues.tpl

```diff
@@ -52,11 +52,30 @@ imagePullSecrets:
 initContainer image logic
 */}}
 {{- define "ocis.initContainerImage" -}}
-  {{- $tag := default .Chart.AppVersion .Values.initContainerImage.tag -}}
+{{- $tag := default "latest" .Values.initContainerImage.tag -}}
+  {{ if $.Values.initContainerImage.sha -}}
+image: "{{ $.Values.initContainerImage.repository }}:{{ $tag }}@sha256:{{ $.Values.initContainerImage.sha }}"
+  {{ else -}}
 image: "{{ $.Values.initContainerImage.repository }}:{{ $tag }}"
+  {{- end }}
 imagePullPolicy: {{ .Values.initContainerImage.pullPolicy }}
 {{- end -}}
 
+{{/*
+jobContainer image logic
+*/}}
+{{- define "ocis.jobContainerImage" -}}
+{{- $tag := default (default "latest" .Values.image.tag) .appSpecificConfig.maintenance.image.tag -}}
+{{- $sha := default .Values.image.sha .appSpecificConfig.maintenance.image.sha -}}
+{{- $repo := default .Values.image.repository .appSpecificConfig.maintenance.image.repository -}}
+  {{ if $sha -}}
+image: "{{ $repo }}:{{ $tag }}@sha256:{{ $sha }}"
+  {{ else -}}
+image: "{{ $repo }}:{{ $tag }}"
+  {{- end }}
+imagePullPolicy: {{ default .Values.image.pullPolicy .appSpecificConfig.maintenance.image.pullPolicy }}
+{{- end -}}
+
```

Comment:
What was failing on your end?

Reply:
In this case the if condition is wrong if I got you write, but this would apply to all our `image` specification right now. You can't achieve something like `owncloud/ocis@sha256:<sha>`. Solution might look like:

```suggestion
{{/*
jobContainer image logic
*/}}
{{- define "ocis.jobContainerImage" -}}
{{- $tag := default (default "latest" .Values.image.tag) .appSpecificConfig.maintenance.image.tag -}}
{{- $sha := default .Values.image.sha .appSpecificConfig.maintenance.image.sha -}}
{{- $repo := default .Values.image.repository .appSpecificConfig.maintenance.image.repository -}}
  {{ if $sha -}}
image: "{{ $repo }}@sha256:{{ $sha }}"
  {{ else -}}
image: "{{ $repo }}:{{ $tag }}"
  {{- end }}
imagePullPolicy: {{ default .Values.image.pullPolicy .appSpecificConfig.maintenance.image.pullPolicy }}
{{- end -}}

```

Does this make sense to you?